tests/translator/input/api_with_custom_domain_route53_multiple.yaml

```diff
@@ -0,0 +1,67 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: >
+  apigateway-2402
+
+  Sample SAM Template for apigateway-2402
+
+Resources:
+            
+  ApiGatewayAdminOne:
+    Type: AWS::Serverless::Api
+    Properties:
+      Name: App-Prod-Web
+      StageName: Prod
+      TracingEnabled: true
+      MethodSettings:
+        - LoggingLevel: Info
+          ResourcePath: '/*'
+          HttpMethod: '*'
+      Domain:
+        DomainName: admin.one.amazon.com
+        CertificateArn: arn::cert::abc
+        EndpointConfiguration: REGIONAL
+        Route53:
+          HostedZoneId: "abc123456"
+      EndpointConfiguration:
+        Type: REGIONAL
+
+            
+  ApiGatewayAdminTwo:
+    Type: AWS::Serverless::Api
+    Properties:
+      Name: App-Prod-Web
+      StageName: Prod
+      TracingEnabled: true
+      MethodSettings:
+        - LoggingLevel: Info
+          ResourcePath: '/*'
+          HttpMethod: '*'
+      Domain:
+        DomainName: admin.two.amazon.com
+        CertificateArn: arn::cert::abc
+        EndpointConfiguration: REGIONAL
+        Route53:
+          HostedZoneId: "abc123456"
+      EndpointConfiguration:
+        Type: REGIONAL
+        
+            
+  ApiGatewayAdminThree:
+    Type: AWS::Serverless::Api
+    Properties:
+      Name: App-Prod-Web
+      StageName: Prod
+      TracingEnabled: true
+      MethodSettings:
+        - LoggingLevel: Info
+          ResourcePath: '/*'
+          HttpMethod: '*'
+      Domain:
+        DomainName: admin.three.amazon.com
+        CertificateArn: arn::cert::abc
+        EndpointConfiguration: REGIONAL
+        Route53:
+          HostedZoneId: "abc123456"
```

Comment:
What happens in `HostedZoneId` is an intrinsic function?

Reply:
The entire `HostedZoneId` is used to generate a logical ID hash of the generated RecordSet Group. For example, consider `HostedZoneId: !Ref MyParam`, `HostedZoneId: !Sub "{{MyParam}}"`, and `HostedZoneId: "abc123456". Even if `MyParam` resolves to `"abc123456"`, these three would yield RecordSet Groups with different logical ID hashes. In the case where only one kind of intrinsics is used for `HostedZoneId`, only one RecordSet Group would be generated.

In the case where multiple kinds of intrinsics are used for `HostedZoneId`, there is no issue at well. Multiple `AWS::Route53::RecordSetGroup` resources can be created in a template pointing to the same `HostedZoneId`. For example, the template below can be deployed without any issue:
```
Resources:
  HostedZone:
    Type: "AWS::Route53::HostedZone"
    Properties:
      Name: "my-website.com"

  MyRecordSetGroup1:
    Type: AWS::Route53::RecordSetGroup
    Properties:
      HostedZoneId: !Ref HostedZone
      RecordSets:
        - Name: "my-website.com"
          Type: "A"
          TTL: "900"
          ResourceRecords:
            - "101.10.21.3"

  MyRecordSetGroup2:
    Type: AWS::Route53::RecordSetGroup
    Properties:
      HostedZoneId: !Ref HostedZone
      RecordSets:
        - Name: "www.my-website.com"
          Type: "CNAME"
          TTL: "900"
          ResourceRecords:
            - "my-website.com"
```